src/library/columncache.h

```diff
@@ -103,12 +103,14 @@ class ColumnCache : public QObject {
         return format.arg(columnNameForFieldIndex(index));
     }
 
-    void insertColumnSortByIndex(
-            int index,
-            const QString& name) {
-        DEBUG_ASSERT(!m_columnSortByIndex.contains(index) ||
-                m_columnSortByIndex[index] == name);
-        m_columnSortByIndex.insert(index, name);
+    void insertColumnSortByEnum(
+            Column column,
+            const QString& sortFormat) {
+        int index = fieldIndex(column);
+        if (index < 0) {
+            return;
+        }
+        m_columnSortByIndex.insert(index, sortFormat);
```

Comment:
The DEBUG_ASSERT got lost

Reply:
We can't debug assert here, because it is normal that we have unused columns returning -1.

Reply:
The deleted debug assert still applies, after verifying that the index is valid and before inserting.